63 mweb select

seangates opened the pull request:
## Description
This set of changes completes the split between `combobox` and `select`, and allows us to allow teams to have a native `select` for their mWeb implementations of their pages/apps.

## References
Closes #63.

seangates commented:
The CI will fail until we publish Skin (either v5, or another pre-release). I'm hoping to release Skin today.

seangates commented:
Okay, Skin v5.0.2 was published yesterday, so this should be working. Let me investigate.

ianmcburnie commented on src/components/ebay-combobox/browser.json:
```diff
@@ -0,0 +1,11 @@
+    "dependencies": [
+        {
+            "if-not-flag": "ebayui-no-skin",
+            "path": "@ebay/skin/listbox"
```
Should this be `@ebay/skin/combobox` instead of `@ebay/skin/listbox `?

ianmcburnie commented on src/components/ebay-combobox/browser.json:
```diff
@@ -0,0 +1,11 @@
+            "if-not-flag": "ebayui-no-skin",
+            "path": "@ebay/skin/listbox"
+        },
+        "../../../src/components/ebay-button",
```
Do we need the dependency on button?

seangates commented:
@ianmcburnie Please have a look. Updated per comments.

Also, I removed the fixed-width `span` from each of the examples for the `ebay-select` as it's not needed.

DylanPiercey commented on src/components/ebay-combobox/index.js:
```diff
@@ -0,0 +1,244 @@
+        selectedOption: state.selected,
+        options: state.options,
+        grow: state.grow,
+        disabled: state.disabled,
```
Disabled does not seem to do anything in this component.

seangates replied:
I think it does here: https://github.com/eBay/ebayui-core/pull/209/files/3fb66418b28ce66c87a6246909619e326ac3844a#diff-854be3f0318a650eed139adb0e266c38R12

DylanPiercey commented on src/components/ebay-combobox/index.js:
```diff
@@ -0,0 +1,244 @@
+    const optionEls = this.el.querySelectorAll(comboboxOptionSelector);
+
+    if (this.state.options && this.state.options.length > 0) {
+        this.expander = new Expander(this.el, {
```
Maybe we should change the expander to be initialized after Marko renders (onUpdate) instead of here.
I think that with this setup if you first render a select with no options, and then again with options it would not work properly.

DylanPiercey replied:
Also if we were to do that we could change it so that the `makeup-expander` is not initialized if the the select is disabled (or empty).

seangates replied:
Yes, we've found that this was an issue. This would clear that up.

DylanPiercey commented on src/components/ebay-combobox/index.js:
```diff
@@ -0,0 +1,256 @@
+}
+
+function onUpdate() {
+    if (!this.expander && this.state.options && this.state.options.length > 0) {
```
This will still break if someone is toggling `disabled` on the component. We need to (onUpdate) check to ensure the the expander is enabled or disabled depending on the current `disabled` state.

seangates replied:
Yeah, I pushed this but am still working on it.

ianmcburnie commented:
Let's not try and fix pre-existing issues with ebay-select in this PR. The scope of this PR is only to rename/reorganise. Let's file them as separate issues for tracking and we will triage them appropriately. 

@seangates Can you please file the two issues we discussed offline.

seangates commented:
Yup. Will do.
